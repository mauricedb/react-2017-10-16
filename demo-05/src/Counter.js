import React, { Component } from "react";

class Counter extends Component {
  render() {
    return (
      <div>
        <div>Count: NaN</div>
        <button>Increment</button>
        <button>Decrement</button>
      </div>
    );
  }
}

export default Counter;
