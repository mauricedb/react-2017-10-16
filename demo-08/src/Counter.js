import React, { Component } from "react";

class Counter extends Component {
  state = {
    count: 0
  };
  
  render() {
    return (
      <div>
        <div>Count: {this.state.count}</div>
        <div />
        <button
          onClick={() => this.setState(state => ({ count: state.count + 1 }))}
        >
          Increment
        </button>
      </div>
    );
  }
}

export default Counter;
