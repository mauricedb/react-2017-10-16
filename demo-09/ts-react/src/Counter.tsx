import * as React from "react";

interface CounterProps {
  label: string;
}

interface CounterState {
  count: number;
}

class Counter extends React.Component<CounterProps, CounterState> {
  state = {
    count: 1
  };
  render() {
    return (
      <div>
        <div>
          {this.props.label} {this.state.count}
        </div>
        <button
          onClick={() => this.setState(state => ({ count: state.count + 1 }))}
        >
          Increment
        </button>
      </div>
    );
  }
}

export default Counter;
