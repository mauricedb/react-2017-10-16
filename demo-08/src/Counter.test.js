import React from "react";
import { shallow } from "enzyme";
import reactTestRenderer from "react-test-renderer";
import Counter from "./Counter";

describe("The Counter component", () => {
  it("renders without crashing", () => {
    shallow(<Counter />);
  });

  it("has a state with count = 0", () => {
    const counter = shallow(<Counter />);
    expect(counter.state()).toEqual({ count: 0 });
  });

  it("has a state with count = 0", () => {
    const counter = shallow(<Counter />);
    expect(counter.state("count")).toBe(0);
  });

  it("has a state with count = 2 after clicking twice", () => {
    const counter = shallow(<Counter />);
    counter.find("button").simulate("click");
    counter.find("button").simulate("click");
    expect(counter.state("count")).toBe(2);
  });

  it("has the same snapshot", () => {
    const json = reactTestRenderer.create(<Counter />).toJSON();

    expect(json).toMatchSnapshot();
  });
});
