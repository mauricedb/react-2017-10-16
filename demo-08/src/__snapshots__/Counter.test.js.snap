// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Counter component has the same snapshot 1`] = `
<div>
  <div>
    Count: 
    0
  </div>
  <div />
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
